cpp/tests/sampling/mg_uniform_neighbor_sampling.cu

```diff
@@ -143,11 +143,33 @@ class Tests_MGUniform_Neighbor_Sampling
 
     rmm::device_uvector<int32_t> batch_number(random_sources.size(), handle_->get_stream());
 
+    auto seed_sizes = cugraph::host_scalar_allgather(
+      handle_->get_comms(), random_sources.size(), handle_->get_stream());
+    size_t num_seeds   = std::reduce(seed_sizes.begin(), seed_sizes.end());
+    size_t num_batches = (num_seeds + uniform_neighbor_sampling_usecase.batch_size - 1) /
+                         uniform_neighbor_sampling_usecase.batch_size;
+
+    std::vector<size_t> seed_offsets(seed_sizes.size());
+    std::exclusive_scan(seed_sizes.begin(), seed_sizes.end(), seed_offsets.begin(), size_t{0});
+
+    thrust::tabulate(
+      handle_->get_thrust_policy(),
+      batch_number.begin(),
+      batch_number.end(),
+      [seed_offset = seed_offsets[handle_->get_comms().get_rank()],
+       num_batches] __device__(int32_t index) { return (seed_offset + index) % num_batches; });
+
+    rmm::device_uvector<int32_t> unique_batches(num_batches, handle_->get_stream());
+    rmm::device_uvector<int32_t> comm_ranks(num_batches, handle_->get_stream());
```

Comment:
So, are we replicating the same `unique_batches` and `comm_ranks` in every GPU?

I might be thinking too far, but this can be a problem if `# labels` is very large. May not need to address in this PR, but something we need to at least start thinking about.